src/utils/utils.js

```diff
@@ -26,3 +37,59 @@ export function createNewDayCells() {
   }
   return ROW_DATA;
 }
+export function calculateSummaryOfSleep(cells) {
+  const reducer = (map, val) => {
+    if (map[val] == null) {
+      map[val] = 1;
+    } else {
+      ++map[val];
+    }
+    return map;
+  };
+  const result = cells.map((cell) => cell.mode).reduce(reducer, {});
+  return result;
+}
+export function clusterCells(cells) {
+  let currentMode = null;
+  let obj = {
+    startIndex: null,
+    endIndex: null,
+    mode: null
+  };
+  const clusteredCells = cells.reduce((result, current, i) => {
+    if (currentMode !== current.mode && currentMode != null) {
+      obj = { ...obj, endIndex: i - 1 };
+      result.push(obj);
+      currentMode = current.mode;
+      obj = {
+        endIndex: null,
+        startIndex: i,
+        mode: current.mode
+      };
+    }
+    if (i === cells.length - 1) {
+      obj = { ...obj, endIndex: i };
+      result.push(obj);
+    }
+    if (currentMode === null) {
+      currentMode = current.mode;
+      obj = { ...obj, startIndex: i, mode: current.mode };
+    }
+
+    return result;
+  }, []);
+
+  return clusteredCells;
+}
+export function calculatePercentage(partialVal, totalVal) {
```

Comment:
question: Are you familiar with [JSDoc](https://jsdoc.app/)? It's a common tool to document javascript functions, especially helpers like in this file. The most common is to document description and arguments. But you can also document `return` and give examples. Like this:

```js
/**
 * Returns percentage of a val based on total
 * @param {Number} partialVal - initial value
 * @param {Number} totalVal - total value
 * @return {Number}
 * @example (1, 2) -> 50
 * @example (2, 8) -> 25
 */
export function calculatePercentage(partialVal, totalVal) {
  // ....
}

```